### ISSUE TITLE
Add depth-capture command: RGBA depth map export with square crop

### ISSUE THREAD

**Issue by arnoldad** [AgrippaRobotics/agrippa-stereocam#19]
## Description

Add a new `depth-capture` command that captures a stereo frame, computes disparity using the classical SGBM backend, converts to depth, and writes an RGBA PNG where:

- **RGB** = camera color (debayered, rectified)
- **A** = normalized depth, `uint8`, scaled to `[0, 255]` against a user-supplied maximum depth

The output is center-cropped to a square and sized to a multiple of 32 pixels, suitable for direct consumption by encoder-decoder neural networks.

For grayscale input (e.g. `--binning 2`), the RGB channels replicate the grayscale value.

## Motivation

ML training pipelines for depth estimation and scene understanding expect dense RGBA or RGB-D images at fixed, square, power-of-two-friendly resolutions. Currently `ag-cam-tools` can preview depth live but cannot export it in a format ready for training datasets. This command closes that gap.

### Existing infrastructure to reuse

| Component | Location | Notes |
|-----------|----------|-------|
| Disparity computation | `src/stereo.h`, `src/stereo_common.c` | `ag_disparity_create()`, `ag_disparity_compute()`, Q4.4 int16 output |
| Depth conversion | `src/stereo.h:137` | `ag_disparity_to_depth(disp_q4, focal_length_px, baseline)` |
| Calibration loading | `src/calib_load.c`, `src/calib_archive.c` | `ag_calib_load()` returns remap tables + `AgCalibMeta` (has `focal_length_px`, `baseline_cm`) |
| DualBayer pipeline | `src/imgproc.c`, `src/image.c` | `extract_dual_bayer_eyes()`, debayer, gamma, remap |
| Image encoding | `src/image.c` | `stbi_write_png()` — currently writes 1-ch or 3-ch; needs 4-ch RGBA path |
| JET colormap | `src/stereo_common.c:249` | `ag_disparity_colorize()` — reference for valid-pixel detection |
| Command dispatch | `src/main.c` | `arg_cmd_register()` pattern |
| Camera config | `src/common.c` | `camera_configure()`, auto-expose, trigger setup |

### Data flow

```
Raw DualBayer frame
  → extract_dual_bayer_eyes (+ software binning if applicable)
  → [Color path]  apply gamma LUT → debayer to RGB24 → rectify
  → [Depth path]  debayer to gray → rectify → ag_disparity_compute (SGBM)
                   → ag_disparity_to_depth per pixel
                   → normalize: alpha = clamp(depth * 255 / max_depth, 0, 255)
                   → invalid pixels (depth ≤ 0): alpha = 0
  → Center-crop both color and alpha to largest inscribed square
  → Resize to target dimension (nearest multiple of 32, or --size N)
  → Composite RGBA: R,G,B from color, A from depth
  → stbi_write_png (4 channels)
```

### Calibration is mandatory

Disparity requires rectified images, and depth conversion requires `focal_length_px` and `baseline_cm`. The command must require either `--calibration-local` or `--calibration-slot`.

### 1. Add RGBA PNG write support to `src/image.h` / `src/image.c`

Add a function:

```c
int write_rgba_png(const char *path,
                   const uint8_t *rgb, const uint8_t *alpha,
                   int width, int height);
```

Interleaves RGB and A into a 4-channel buffer, calls `stbi_write_png()` with `comp=4`.

### 2. Add square-crop + resize helpers to `src/imgproc.h` / `src/imgproc.c`

```c
/* Center-crop to the largest inscribed square, return new side length. */
void crop_center_square(const uint8_t *src, int src_w, int src_h, int channels,
                        uint8_t *dst, int *side);

/* Nearest-neighbor resize (sufficient for depth maps). */
void resize_nn(const uint8_t *src, int src_side, int channels,
               uint8_t *dst, int dst_side);

/* Round down to nearest multiple of 32. */
static inline int round_down_32(int x) { return x & ~31; }
```

### 3. Create `src/cmd_depth_capture.c`

New command file following the pattern of `cmd_capture.c` and `cmd_depth_preview.c`.

**CLI arguments:**

| Flag | Description |
|------|-------------|
| `-s`, `--serial` | Camera serial |
| `-a`, `--address` | Camera IP |
| `-i`, `--interface` | Force NIC |
| `-o`, `--output` | Output directory (default: `.`) |
| `-x`, `--exposure` | Exposure µs |
| `-g`, `--gain` | Gain dB |
| `-A`, `--auto-expose` | Auto-expose then lock |
| `-b`, `--binning` | Sensor binning: `1` or `2` |
| `-p`, `--packet-size` | GigE packet size |
| `--calibration-local` | Calibration directory (required, mutually exclusive with slot) |
| `--calibration-slot` | On-camera slot 0-2 (required, mutually exclusive with local) |
| `--max-depth` | Maximum depth for normalization, in cm (required) |
| `--size` | Output side length in px, must be multiple of 32 (default: auto) |
| `-v`, `--verbose` | Diagnostic output |

**Logic:**

1. Parse + validate args. Require calibration. Require `--max-depth`.
2. Connect, configure camera (single-frame trigger, same as `cmd_capture`).
3. Load calibration (remap tables + meta). Verify `focal_length_px > 0` and `baseline_cm > 0`.
4. If `--auto-expose`: settle exposure, lock.
5. Capture one frame.
6. Deinterleave DualBayer → left/right raw.
7. **Color path:** gamma → debayer to RGB (or gray-replicate if binned) → rectify → `rect_rgb_left`.
8. **Depth path:** debayer to gray → rectify → `ag_disparity_compute()` → `disparity_buf` (Q4.4 int16).
9. Per-pixel depth: `ag_disparity_to_depth()` → normalize to uint8 alpha.
10. Center-crop `rect_rgb_left` (3-ch) and `alpha_buf` (1-ch) to square.
11. If `--size` given: resize both to that dimension. Otherwise: `round_down_32(square_side)`.
12. `write_rgba_png(output_path, cropped_rgb, cropped_alpha, side, side)`.
13. Print output path and dimensions.

### 4. Register the command in `src/main.c`

```c
int cmd_depth_capture (int argc, char *argv[], arg_dstr_t res, void *ctx);
// ...
arg_cmd_register ("depth-capture", cmd_depth_capture,
                  "Capture RGBA depth map (RGB + depth alpha)", NULL);
```

### 5. Update `Makefile`

Add `$(SRCDIR)/cmd_depth_capture.c` to `SRCS`.

### 6. Shell completions

Add `depth-capture` and its flags to `completions/ag-cam-tools.bash` and `.zsh`.

### Unit tests

- `crop_center_square()`: verify crop offset and output dimensions for landscape, portrait, and already-square inputs.
- `round_down_32()`: edge cases — 32, 33, 63, 64, 0.
- `resize_nn()`: identity (same size), 2× downscale, non-power-of-two.
- `write_rgba_png()`: round-trip — write then read back with `stbi_load`, verify 4 channels, verify alpha values.

### Hardware integration tests

Following `test_capture_rectify_hw.sh` pattern:

1. `depth-capture -a <IP> --calibration-local <dir> --max-depth 500 -x 50000` — produces RGBA PNG, verify 4-channel with `identify` or `file`.
2. `depth-capture ... --size 640` — verify output is 640×640.
3. `depth-capture ... --size 641` — rejected (not multiple of 32).
4. Missing `--calibration-*` — rejected with clear error.
5. Missing `--max-depth` — rejected with clear error.
6. `depth-capture ... -A --max-depth 300` — auto-expose + depth succeeds.
7. `depth-capture ... -b 2 --max-depth 500` — binned mode produces grayscale-replicated RGB.

## Documentation Updates

- Create `docs/cli/depth-capture.md` with examples, option table, and explanation of the RGBA format.
- Add `depth-capture` row to `docs/cli/overview.md` commands table.
- Link `depth-capture.md` from `docs/cli/overview.md` command pages list.
- Add `depth-capture` to `docs/index.md` main capabilities.
- Add the page to `mkdocs.yml` nav.

## Design Decisions

- **8-bit alpha only.** 256 depth bins are sufficient for the target ML use case. A 16-bit path can be added later if needed.
- **SGBM only (initially).** Neural backends can be added via a `--backend` flag in a follow-up.
- **Center crop.** Simplest strategy for square extraction. An ROI-based crop could be added later.
- **Nearest-neighbor resize.** Avoids interpolation artifacts in the depth channel. Bilinear could be offered as an option if color quality matters.
- **`alpha = 0` for invalid depth.** Consistent with the convention that transparent pixels have no depth data. Downstream consumers can mask on `A == 0`.
